### PR TITLE
MODOAIPMH-283 provide effective location and effective call number data when item record is not present

### PR DESCRIPTION
**Pull request by obozhko-folio**
Jira: https://issues.folio.org/browse/MODOAIPMH-283

### PURPOSE

The existing FOLIO implementation of the derived fields (effective location and effective call number) requires item records. The purpose of this story is to populate location and holdings information, following the logic of effective location/call number when the item data is not present.

### CODE REVIEW

**Review by alekGbuz — APPROVED** [review]

File: src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java [hunk 1]
```diff
@@ -488,14 +506,20 @@ private void getNextBatch(String requestId, Request request, boolean firstBatch,
       try {
         for (JsonEvent jsonEvent : batch) {
           JsonObject value = jsonEvent.objectValue();
-          String instanceId = value.getString(ENRICHED_INSTANCE_ID);
-          Object itemsandholdingsfields = value.getValue(RecordMetadataManager.ITEMS_AND_HOLDINGS_FIELDS);
+          String instanceId = value.getString(INSTANCE_ID_FIELD_NAME);
+          Object itemsandholdingsfields = value;
```
Comment (khandramai) [inline src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java:510]: Why this variable is Object type if value-variable is JsonObject type?

File: src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java [hunk 2]
```diff
@@ -488,14 +506,20 @@ private void getNextBatch(String requestId, Request request, boolean firstBatch,
...
           if (itemsandholdingsfields instanceof JsonObject) {
```
Comment (khandramai) [inline src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java:511]: itemsandholdingsfields equals to variable value. Why do we need to have two the same variables?

File: src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java (hunk 1 above)
Comment (obozhko-folio) [reply src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java]: Yes, agree. Moreover, this variable is redundant.

File: src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java (hunk 2 above)
Comment (obozhko-folio) [reply src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java]: You are right, the redundant variable has been removed.

**Review by khandramai — APPROVED** [review]

**Pull request merged**